ruby/examples/lib/register_user_with_form.rb

```diff
@@ -0,0 +1,39 @@
+require 'active_model'
+
+class RegisterUserWithForm
+  Request = Struct.new(:email, :name, :password)
+  Response = Struct.new(:form)
+
+  class Response::Form
+    include ActiveModel::Model
+
+    attr_reader *Request.members
+    attr_reader :errors
+
+    def initialize(email:, name:, password:, errors:)
+      @email = email
+      @name = name
+      @password = password
+      @errors = errors
+    end
+  end
+
+  attr_reader :gateway
+
+  def initialize(gateway: Gateway.new)
+    @gateway = gateway
+  end
+
+  def register(request)
+    errors = gateway.create_user(request.to_h)
+    form = Response::Form.new(request.to_h.merge(errors: errors))
```

Comment:
I would never return a form from a usecase - this makes the use case entirely immobile and hangs a source code dependency that points outward from the use case to the delivery mechanism. 

This creates Viscosity and creates Rigidity... as the delivery mechanism's views may need to change if the use case changes